src/VisualStudio/Core/SolutionExplorerShim/GeneratedSourceItem/GeneratedSourceFolderItemProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.Internal.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
+{
+    [Export(typeof(IAttachedCollectionSourceProvider))]
+    [Name("GeneratedSourceFolderProvider")]
+    [Order(Before = HierarchyItemsProviderNames.Contains)]
+    internal sealed class GeneratedSourceFolderItemProvider : ProjectFolderItemProvider
+    {
+        [ImportingConstructor]
+        public GeneratedSourceFolderItemProvider(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider) :
+            base(serviceProvider)
+        {
+        }
+
+        protected override IAttachedCollectionSource CreateCollectionSource(IVsHierarchyItem item, string relationshipName)
+        {
+            if (item != null &&
+                item.Parent != null &&
+                item.Parent.Parent == null)
```

Comment:
The check ignores items where the parent is not a top-level (project) item. Added a comment.